.github/workflows/create-release.yml

```diff
@@ -0,0 +1,40 @@
+name: Create new release
+
+on:
+  # run when a release is published
+  # triggers for both prerelease and release
+  release:
+    types:
+    - published
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0 # required to access tags
+          submodules: 'true'
+
+      - name: Build & run devcontainer image
```

Comment:
Above with the "build and push Devcontainer image" task in `build-devcontainer-image.yml` we use the `docker/build-push-action@v2`. It seems that has a `push` option (which could be set to false?)

On the other hand I see in `pr-validation` we do basically this what you're doing here.

A couple questions about this:
1. Should we be using that `build-and-push` task more universally?
2. If the answer to the above is no, should we maybe have a comment here and in `pr-validation` that basically says "these two things should be kept in sync"?

Reply:
The problem is that we need things to happen inside the devcontainer. Now, Github has [support for running jobs in a container](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idcontainer) but we need the container to really be “cached container image + potential updates to the Dockerfile” so that you can make changes in a PR, and I can’t see a way to achieve that.

Reply:
Maybe if I have two jobs, one that builds an image and the other one that uses it…

Reply:
Don't go out of your way to do a big change for this. I was just curious about the two approaches is all. This is fine and in some ways "better" because it's not opaque and we can see what is happening.

Reply:
…but that’s going to be tricky, would have to push each image to the registry.

Reply:
Going to leave this for now :)